PIPs falsely ignored by the int_generate.py script.

One of the CI runs (https://source.cloud.google.com/results/invocations/b5621a37-f0c2-43fb-a4c1-f8a8d50ef71c/targets) revealed an issue in the int_generate.py script causing the 056-pip-rem fuzzer to fail occasionally with the following message:
<pre>
Compiling segment data.
Traceback (most recent call last):
  File "/home/tmichalak/Development/X-Ray/prjxray2/fuzzers/int_generate.py", line 94, in <module>
    segmk.write()
  File "/home/tmichalak/Development/X-Ray/prjxray2/prjxray/segmaker.py", line 399, in write
    ]) != 0, "Didn't  generate any segments"
AssertionError: Didn't  generate any segments
</pre>

Problem description:
In the failing test case we had the following PIP in the todo list:
<pre>
INT_R.LH0.SS6END3
</pre>

whereas the design, read by the int_generate.py script from the generated design.txt had among others the same PIP but on an INT_L site:
<pre>
INT_R.LH0.SS6END3
INT_L.LH0.SS6END3
</pre>

The script implements an ignoring mechanism for PIPs that are not in the todo list, however it only adds the src and dst wires to the ignore pip list, without the tile information.
This leads to falsely adding a PIP present in the todo list to the list of ignored PIPs.

Solution:
Obviously, we also need to keep the information whether the PIP is on the INT_L or INT_R tile up to the point where the ignoring mechanism comes in.

--- a/fuzzers/int_generate.py
+++ b/fuzzers/int_generate.py
@@ -54,10 +54,11 @@ with open(args.design, "r") as f:
         if tile not in tiledata:
             tiledata[tile] = {"pips": set(), "srcs": set(), "dsts": set()}
 
-        if pip in pipdata:
-            assert pipdata[pip] == (src, dst)
+        tile_lr = ("_".join(tile.split("_")[0:2]))
+        if (tile_lr, pip) in pipdata:
+            assert pipdata[(tile_lr, pip)] == (src, dst)
         else:
-            pipdata[pip] = (src, dst)
+            pipdata[(tile_lr, pip)] = (src, dst)
 
         tiledata[tile]["pips"].add(pip)
         tiledata[tile]["srcs"].add(src)
@@ -67,23 +68,23 @@ with open(args.design, "r") as f:
             tiledata[tile]["srcs"].add(dst)
             tiledata[tile]["dsts"].add(src)
 
+        t = ("_".join(tile.split("_")[0:2]), dst, src)
         if pnum == 1 or pdir == 0:
             verbose and print('ignore pnum == 1 or pdir == 0: ', pip)
-            ignpip.add(pip)
+            ignpip.add(t)
 
-        t = ("_".join(tile.split("_")[0:2]), dst, src)
         if t not in todo:
             verbose and print('ignore not todo: ', t)
-            ignpip.add(pip)
+            ignpip.add(t)
 
 for tile, pips_srcs_dsts in tiledata.items():
     pips = pips_srcs_dsts["pips"]
     srcs = pips_srcs_dsts["srcs"]
     dsts = pips_srcs_dsts["dsts"]
-
-    for pip, src_dst in pipdata.items():
+    for (tile_lr, pip), src_dst in pipdata.items():
         src, dst = src_dst
-        if pip in ignpip:
+        t = (tile_lr, dst, src)
+        if t in ignpip:
             pass
         elif pip in pips:
             segmk.add_tile_tag(tile, "%s.%s" % (dst, src), 1)

